cartographer/mapping_2d/global_trajectory_builder.cc

```diff
@@ -35,6 +36,17 @@ const Submaps* GlobalTrajectoryBuilder::submaps() const {
 void GlobalTrajectoryBuilder::AddRangefinderData(
     const common::Time time, const Eigen::Vector3f& origin,
     const sensor::PointCloud& ranges) {
+  // TODO average laser scans instead of discarding them?
```

Comment:
Not sure what you mean by averaging, but would it also help in your case if you combine multiple scans using the predicted poses in the `LocalTrajectoryBuilder` introducing a parameter `scans_per_accumulation`? This would be similar to the 3D case if you look at the `mapping_3d/kalman_local_trajectory_builder.cc` for example.

Otherwise we now add a stop gap solution that we will remove again in the future, instead of moving a smaller step in the direction we would like to go.